astropy/coordinates/representation.py

```diff
@@ -42,8 +42,12 @@ def __init__(cls, name, bases, dct):
         if cls.__name__ == 'BaseRepresentation':
             return
 
-        REPRESENTATION_CLASSES[cls.get_name()] = cls
+        repr_name = cls.get_name()
 
+        if repr_name in REPRESENTATION_CLASSES:
+            raise ValueError("Representation class {0} already defined".format(repr_name))
+
+        REPRESENTATION_CLASSES[repr_name] = cls
```

Comment:
Yes, I was thinking that. I don't see any good reason to override existing ones as it's hard to predict side-effects.